homeassistant/components/poolsense/sensor.py

```diff
@@ -14,101 +16,85 @@
-    },
-    "pH High": {
-        "unit": None,
-        "icon": "mdi:pool",
-        "name": "pH High",
-        "device_class": None,
-    },
-    "pH Low": {
-        "unit": None,
-        "icon": "mdi:pool",
-        "name": "pH Low",
-        "device_class": None,
-    },
-}
+SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key="Chlorine",
+        native_unit_of_measurement=ELECTRIC_POTENTIAL_MILLIVOLT,
+        icon="mdi:pool",
+        name="Chlorine",
+    ),
+    SensorEntityDescription(
+        key="pH",
+        icon="mdi:pool",
+        name="pH",
+    ),
+    SensorEntityDescription(
+        key="Battery",
+        native_unit_of_measurement=PERCENTAGE,
+        name="Battery",
+        device_class=DEVICE_CLASS_BATTERY,
+    ),
+    SensorEntityDescription(
+        key="Water Temp",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        icon="mdi:coolant-temperature",
+        name="Temperature",
+        device_class=DEVICE_CLASS_TEMPERATURE,
+    ),
+    SensorEntityDescription(
+        key="Last Seen",
+        icon="mdi:clock",
+        name="Last Seen",
+        device_class=DEVICE_CLASS_TIMESTAMP,
+    ),
+    SensorEntityDescription(
+        key="Chlorine High",
+        native_unit_of_measurement=ELECTRIC_POTENTIAL_MILLIVOLT,
+        icon="mdi:pool",
+        name="Chlorine High",
+    ),
+    SensorEntityDescription(
+        key="Chlorine Low",
+        native_unit_of_measurement=ELECTRIC_POTENTIAL_MILLIVOLT,
+        icon="mdi:pool",
+        name="Chlorine Low",
+    ),
+    SensorEntityDescription(
+        key="pH High",
+        icon="mdi:pool",
+        name="pH High",
+    ),
+    SensorEntityDescription(
+        key="pH Low",
+        icon="mdi:pool",
+        name="pH Low",
+    ),
+)
 
 
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Defer sensor setup to the shared sensor module."""
     coordinator = hass.data[DOMAIN][config_entry.entry_id]
 
-    sensors_list = []
-    for sensor in SENSORS:
-        sensors_list.append(
-            PoolSenseSensor(coordinator, config_entry.data[CONF_EMAIL], sensor)
-        )
+    entities = [
+        PoolSenseSensor(coordinator, config_entry.data[CONF_EMAIL], description)
+        for description in SENSOR_TYPES
+    ]
 
-    async_add_entities(sensors_list, False)
+    async_add_entities(entities, False)
 
 
 class PoolSenseSensor(PoolSenseEntity, SensorEntity):
     """Sensor representing poolsense data."""
 
-    @property
-    def name(self):
-        """Return the name of the particular component."""
-        return f"PoolSense {SENSORS[self.info_type]['name']}"
+    _attr_extra_state_attributes = {ATTR_ATTRIBUTION: ATTRIBUTION}
+
+    def __init__(self, coordinator, email, description: SensorEntityDescription):
+        """Initialize PoolSense sensor."""
+        super().__init__(coordinator, email, description)
+        self._attr_name = f"PoolSense {description.name}"
```

Comment:
If this is set in `PoolSenseEntity` there is no need to add these init methods to `PoolSenseSensor` and `PoolSenseBinarySensor`